appinventor/blocklyeditor/src/blocks/components.js

```diff
@@ -241,7 +255,7 @@ Blockly.Blocks.component_event = {
   },
 
   declaredNames: function() { // [lyn, 10/13/13] Interface with Blockly.LexicalVariable.renameParam
-    return this.getVars();
+    return this.isGeneric ? [this.getFieldValue('VAR0')].concat(this.getVars()) : this.getVars();
```

Comment:
Because the general event adds a function parameter in blocky, but this parameter is not defined in source code. Additional parameter is not recognized elsewhere using getVars and getVarString.I don't want to change everything that uses getVars.